modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java

```diff
@@ -1428,6 +1431,101 @@ public void testAddObjectEntryWithAttachmentObjectField() throws Exception {
 			objectDefinition.getObjectDefinitionId());
 	}
 
+	@Test
+	public void testAddOrUpdateObjectEntryWithPicklistObjectField()
+		throws Exception {
+
+		ListTypeDefinition listTypeDefinition =
+			listTypeDefinitionLocalService.addListTypeDefinition(
+				null, TestPropsValues.getUserId(),
+				Collections.singletonMap(
+					LocaleUtil.getDefault(), RandomTestUtil.randomString()),
+				false,
+				ListUtil.fromArray(
+					ListTypeEntryUtil.createListTypeEntry(
+						"listTypeEntryKey1",
+						Collections.singletonMap(
+							LocaleUtil.US, RandomTestUtil.randomString())),
+					ListTypeEntryUtil.createListTypeEntry(
+						"listTypeEntryKey2",
+						Collections.singletonMap(
+							LocaleUtil.US, RandomTestUtil.randomString()))));
+
+		ObjectDefinition objectDefinition = _createObjectDefinition(
+			Collections.singletonList(
+				new PicklistObjectFieldBuilder(
+				).indexed(
+					true
+				).labelMap(
+					LocalizedMapUtil.getLocalizedMap(
+						RandomTestUtil.randomString())
+				).listTypeDefinitionId(
+					listTypeDefinition.getListTypeDefinitionId()
+				).name(
+					"picklistObjectFieldName"
+				).build()));
+
+		try {
+			long objectEntryId1 =
+				_addAndAssertObjectEntryWithPicklistObjectField(
+					null, null, objectDefinition);
+			long objectEntryId2 =
+				_addAndAssertObjectEntryWithPicklistObjectField(
+					StringPool.BLANK, StringPool.BLANK, objectDefinition);
+			long objectEntryId3 =
+				_addAndAssertObjectEntryWithPicklistObjectField(
+					"listTypeEntryKey1", "listTypeEntryKey1", objectDefinition);
+			long objectEntryId4 =
+				_addAndAssertObjectEntryWithPicklistObjectField(
+					"listTypeEntryKey2", "listTypeEntryKey2", objectDefinition);
+
+			ObjectField objectField = _objectFieldLocalService.fetchObjectField(
+				objectDefinition.getObjectDefinitionId(),
+				"picklistObjectFieldName");
+
+			_objectFieldSettingLocalService.addObjectFieldSetting(
+				TestPropsValues.getUserId(), objectField.getObjectFieldId(),
+				ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
+				"listTypeEntryKey1");
+			_objectFieldSettingLocalService.addObjectFieldSetting(
+				TestPropsValues.getUserId(), objectField.getObjectFieldId(),
+				ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+				ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE);
+
+			Assert.assertEquals(
+				StringPool.BLANK,
+				_getListEntryKey(
+					_defaultObjectEntryManager.getObjectEntry(
+						_simpleDTOConverterContext, objectDefinition,
+						objectEntryId2)));
+
+			_updateAndAssertObjectEntryWithPicklistObjectField(
+				null, null, objectDefinition, objectEntryId1);
```

Comment:
I thought that this case would return the default value :/

Reply:
we only add the default value when creating a new entry